### ISSUE TITLE
Issue with patchcore anomaly score for large confidence scores

### ISSUE THREAD

**Issue by bsl546** [openvinotoolkit/anomalib#324]
**Describe the bug**
When the confidence is very high for images treated by the patchcore model, the anomaly score can be nan.
This is due to the formula:
weights = 1 - (torch.max(torch.exp(confidence)) / torch.sum(torch.exp(confidence)))
in anomaly_map.py, torch.exp() gives nan for values too large.

**Expected behavior**
Two possible solutions: 
- Either  a threshold should be set on the tensors, like
confidence[confidence>thres] = thres
Possible values for thres are e.g., 50.0.

- Or, even cleaner, replace 1- (torch.max(torch.exp(confidence)) / torch.sum(torch.exp(confidence))) by 1-torch.max(torch.nn.Softmax()(confidence))

**Comment by bsl546** [comment]
Or one should use softmax instead, that works as well.

**Comment by djdameln** [comment]
Hi, thanks for spotting this and apologies for the late response. I agree that the latter solution, replacing by `1 - torch.max(torch.nn.Softmax()(confidence))`, would be the best way to address this. Would you be interested in submitting a PR for this so you could become a contributor?

**Comment by bsl546** [comment]
Yes sure, I will do it today if possible.
